src/elements/content-sidebar/activity-feed/task-new/Task.js

```diff
@@ -345,38 +368,53 @@ class Task extends React.Component<Props, State> {
                                 users={isAssigneeListOpen ? assignedToFull : assigned_to}
                             />
                         </div>
-                        {currentUserAssignment && shouldShowActions && (
+                        {shouldShowActions && (
                             <div className="bcs-Task-actionsContainer">
-                                <TaskActions
-                                    taskType={task_type}
-                                    onTaskApproval={
-                                        isPending
-                                            ? noop
-                                            : () => {
-                                                  this.handleTaskAction(
-                                                      id,
-                                                      currentUserAssignment.id,
-                                                      TASK_NEW_APPROVED,
-                                                  );
+                                {isMultiFile
+                                    ? onView && (
+                                          <PrimaryButton
+                                              onClick={() => onView(id, isCreator)}
+                                              data-testid="view-details-button"
+                                          >
+                                              <FormattedMessage {...messages.tasksFeedViewDetailsAction} />
+                                          </PrimaryButton>
+                                      )
+                                    : currentUserAssignment && (
+                                          <TaskActions
```

Comment:
What do you think about either:
- putting the multi-file action inside TaskActions, because it already has logic to decide what buttons to show (Approve/Reject | Complete)
- breaking apart TaskActions to keep the logic in one place

Reply:
I'll move it into TaskActions